studio/components/interfaces/Functions/EdgeFunctionDetails.tsx

```diff
@@ -0,0 +1,332 @@
+import dayjs from 'dayjs'
+import { useRouter } from 'next/router'
+import { FC, useState, useEffect } from 'react'
+import * as Tooltip from '@radix-ui/react-tooltip'
+import { PermissionAction } from '@supabase/shared-types/out/constants'
+import {
+  Alert,
+  IconGlobe,
+  IconTerminal,
+  IconMinimize2,
+  IconMaximize2,
+  IconCheck,
+  IconClipboard,
+  Button,
+  Modal,
+} from 'ui'
+
+import { useStore, useParams, checkPermissions } from 'hooks'
+import Panel from 'components/ui/Panel'
+import CommandRender from './CommandRender'
+import { useProjectApiQuery } from 'data/config/project-api-query'
+import { useEdgeFunctionDeleteMutation } from 'data/edge-functions/edge-functions-delete-mutation'
+
+interface Props {}
+
+// [Joshen] Next - additional configs: Verify jwt + import maps
+
+const EdgeFunctionDetails: FC<Props> = () => {
+  const router = useRouter()
+  const { functions, ui } = useStore()
+  const { ref: projectRef, id } = useParams()
+  const [isCopied, setIsCopied] = useState(false)
+  const [showDeleteModal, setShowDeleteModal] = useState(false)
+  const [showInstructions, setShowInstructions] = useState(false)
+  const [selectedFunction, setSelectedFunction] = useState<any>(null)
+
+  const canDeleteEdgeFunction = checkPermissions(PermissionAction.FUNCTIONS_WRITE, '*')
+  const { mutateAsync: deleteEdgeFunction, isLoading: isDeleting } = useEdgeFunctionDeleteMutation()
+
+  useEffect(() => {
+    setSelectedFunction(functions.byId(id))
+  }, [functions.isLoaded, ui.selectedProject])
+
+  const { data: settings } = useProjectApiQuery({ projectRef })
+
+  // Get the API service
+  const apiService = settings?.autoApiService
+  const anonKey = apiService?.service_api_keys.find((x) => x.name === 'anon key')
+    ? apiService.defaultApiKey
+    : undefined
+
+  const endpoint = apiService?.app_config.endpoint ?? ''
+  const endpointSections = endpoint.split('.')
+  const functionsEndpoint = [
+    ...endpointSections.slice(0, 1),
+    'functions',
+    ...endpointSections.slice(1),
+  ].join('.')
+  const functionUrl = `https://${functionsEndpoint}/${selectedFunction?.slug}`
```

Comment:
It's kind of irrelevant at the moment, as edge functions aren't on self-hosted, but you could specify `apiService.protocol` instead of hardcoding `https`. I'll leave this one up to you though.